### PR TITLE
Post game stats pages

### PR DESCRIPTION
**Pull request by jasminetai**
Postgame stats pages for player and host.

For the player, they see a pie chart with data from their correct/wrong/unanswered questions and a label ranking (Perfect/Awesome/Great/Nice/Good try) based on the percentage of right answers.

For the host, they see a histogram of percentages of right answers along with the class's average percentage. The histogram currently operates on the counts of percentages seen, rounded to the nearest whole number. It was a little tricky to make the histogram responsive, although we do not need to worry about mobile host views (and I don't know how the histogram can be compacted into a phone screen's dimentions anyway...). Feedback on the histogram would be especially useful.

### CODE REVIEW

File: frontend/pages/[gameId]/postgameplayer.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+					className="-mb-4"
+				/>
+
+				<div className="w-fit bg-gray-100 border border-black px-8 py-1 text-3xl z-20 2xl:text-4xl">
+					How you did
+				</div>
+
+				<div className="w-full h-3/4 bg-purple-500 flex items-center justify-center p-8 pt-10 -mt-6 text-xl rounded-3xl z-10 sm:h-2/3 sm:p-12 sm:rounded-xl 2xl:text-3xl">
+					<div className="w-full h-full flex flex-col items-center justify-center gap-8 bg-gray-100 bg-opacity-50 rounded-xl shadow-heavy sm:flex-row sm:gap-16 sm:w-11/12 2xl:gap-20">
+						<PieChart
```
Comment (celdond) [inline frontend/pages/[gameId]/postgameplayer.tsx:51]: The css for the PieChart is mostly good, but there are sizes before sm where it essentially vanishes.

File: frontend/pages/[gameId]/postgameplayer.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+				>
+					Quit
+				</Link>
+				<button className="grow bg-purple-50 self-end px-8 py-2 rounded-lg text-lg text-white text-center shadow-heavy hover:brightness-110 2xl:text-xl sm:absolute sm:top-6 sm:right-48 2xl:right-52">
+					Export Quiz
+				</button>
+			</div>
+			<button className="w-3/4 self-center bg-purple-50 self-end px-8 py-2 mb-2 rounded-lg text-lg text-white text-center shadow-heavy hover:brightness-110 sm:absolute sm:top-6 sm:right-6 sm:w-fit 2xl:text-xl">
+				Play Again
+			</button>
```
Comment (celdond) [inline frontend/pages/[gameId]/postgameplayer.tsx:106]: These buttons can end up partially obscured on shorter sm sized windows/screens.

**Review by celdond — APPROVED** [review]
Logic is solid, but there are some small css adjustments that can be made that I marked.  These are definitely edge cases, but I do think the pie chart should be adjusted a bit before merging.
Regardless, I would suggest waiting until the websocket integration is done before merging this so we can include that logic in this branch, but it wouldn't be a problem to merge it as is.

Additional note: the histogram is fine imo, it is a little cramped in smaller dimensions but it is nothing I would suggest to fix.

File: frontend/pages/[gameId]/postgameplayer.tsx (hunk 1 above)
Comment (jasminetai) [reply frontend/pages/[gameId]/postgameplayer.tsx:51]: Good catches, I'll definitely fix them!

File: frontend/components/QuestionPages/QuestionAnswers.tsx
Comment (190n) [inline frontend/components/QuestionPages/QuestionAnswers.tsx:1]: I've significantly restructured this file on my branch but I'll integrate these color changes.

File: frontend/pages/[gameId]/postgameplayer.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+						/>
+
+						<div className="flex flex-col items-center justify-center gap-4 sm:gap-6 2xl:gap-10">
+							<div className="w-fit bg-green-200 px-2 py-1 rounded-lg text-white shadow-heavy 2xl:px-4 2xl:py-2">
+								Correct: {postgameData.correct}
+							</div>
+							<div className="w-fit bg-red-200 px-2 py-1 rounded-lg text-white shadow-heavy 2xl:px-4 2xl:py-2">
+								Wrong: {postgameData.wrong}
+							</div>
+							<div className="w-fit bg-yellow-200 px-2 py-1 rounded-lg text-gray-200 shadow-heavy 2xl:px-4 2xl:py-2">
```
Comment (190n) [inline frontend/pages/[gameId]/postgameplayer.tsx]: Contrast on all these texts is a little low -- what if we made them all white but with a text shadow, or all black?

**Review by 190n — COMMENTED** [review]
See minor comments, otherwise this all looks great. Good job.

File: frontend/pages/[gameId]/postgameplayer.tsx (hunk 3 above)
Comment (jasminetai) [reply frontend/pages/[gameId]/postgameplayer.tsx]: I see what you mean, I will try all black.

**Pull request merged**